ontology/rml-core.owl

```diff
@@ -0,0 +1,517 @@
+vann:preferredNamespaceUri rdf:type owl:AnnotationProperty .
+
+
+###  http://www.linkedmodel.org/schema/vaem#acronym
+vaem:acronym rdf:type owl:AnnotationProperty .
+
+
+###  http://xmlns.com/foaf/0.1/name
+foaf:name rdf:type owl:AnnotationProperty .
+
+
+#################################################################
+#    Object Properties
+#################################################################
+
+###  http://w3id.org/rml/class
+:class rdf:type owl:ObjectProperty ;
+       rdfs:domain :SubjectMap ;
+       rdfs:range rdfs:Class ;
+       rdfs:comment "The subject value generated for a logical table row will be asserted as an instance of this RDFS class."@en ;
+       rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+       rdfs:label "class" .
+
+
+###  http://w3id.org/rml/constant
+:constant rdf:type owl:ObjectProperty ;
+          rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+          rdfs:label "constant" .
+
+
+###  http://w3id.org/rml/datatype
+:datatype rdf:type owl:ObjectProperty ;
+          rdfs:domain :ObjectMap ;
+          rdfs:range rdfs:Datatype ;
+          rdfs:comment "Specifies the datatype of the object component for the generated triple from a logical table row."@en ;
+          rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+          rdfs:label "datatype" .
+
+
+###  http://w3id.org/rml/datatypeMap
+:datatypeMap rdf:type owl:ObjectProperty ;
+             rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+             rdfs:label "datatype map" .
+
+
+###  http://w3id.org/rml/graph
+:graph rdf:type owl:ObjectProperty ;
+       rdfs:domain [ rdf:type owl:Class ;
+                     owl:unionOf ( :PredicateObjectMap
+                                   :SubjectMap
+                                 )
+                   ] ;
+       rdfs:comment "An IRI reference for use as the graph name of all triples generated with the GraphMap."@en ;
+       rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+       rdfs:label "graph" .
+
+
+###  http://w3id.org/rml/graphMap
+:graphMap rdf:type owl:ObjectProperty ;
+          rdfs:domain [ rdf:type owl:Class ;
+                        owl:unionOf ( :PredicateObjectMap
+                                      :SubjectMap
+                                    )
+                      ] ;
+          rdfs:range :GraphMap ;
+          rdfs:comment "Specifies a GraphMap. When used with a SubjectMap element, all the RDF triples generated from a logical row will be stored in the specified named graph. Otherwise, the RDF triple generated using the (predicate, object) pair will be stored in the specified named graph."@en ;
+          rdfs:isDefinedBy <http://w3id.org/rml/core/> ;
+          rdfs:label "graph map" .
+
+
+###  http://w3id.org/rml/joinCondition
+:joinCondition rdf:type owl:ObjectProperty ;
+               rdfs:domain [ rdf:type owl:Class ;
+                             owl:unionOf ( :RefObjectMap
+                                           :StarMap
```

Comment:
This shouldn't be declared in core, right?

Reply:
Since `:joinCondition` belongs to the core, I put all its correspondent "restriction" there; it only extends the domain of the property, is not forcing to use it in any case. I've done also the same for the properties in the core that point to resources in other modules (e.g. range of `:logicalSource` --`:LogicalSource`-- that now belongs in the Source&Target spec

Reply:
I would prefer to completely decouple the ontologies, otherwise managing these connections becomes quite a pain.

Do we actually need to specify this domain restriction at all? There isn't really any useful inference to be made from it IMO.
I would prefer to remove the restriction over having elements from other modules leak into core.

Reply:
If we still really want to express the domain of `:joinCondition`, a more extensible approach would be to define a new class (something like) `:JoiningElement` and use that as the `rdfs:domain` of  `:joinCondition`. Then have `:RefObjectMap` be a sub-class of `:JoiningElement` and other modules can then add their own sub-class relationships.
This way the core module can stay independent, and new modules can add their own extensions without impacting the core model.

Reply:
I see your point; then I'd rather remove the restriction and avoid adding more elements that may add more noise in the end

Reply:
@andimou we're leaning into dropping this restrictions completely, i.e. not specifying any domain in this case; what's your take on this?